src/python_src/util/app_utilities.py

```diff
@@ -122,6 +123,76 @@ def load_config(config_file: str) -> Dict[str, Any]:
 )
 
 
+def calculate_file_sha256(file_path: str, chunk_size: int = 4096) -> str:
+    """Calculate SHA-256 hash of a file."""
+    sha256_hash = hashlib.sha256()
+    try:
+        with open(file_path, "rb") as f:
+            for chunk in iter(lambda: f.read(chunk_size), b""):
+                sha256_hash.update(chunk)
+        return sha256_hash.hexdigest()
+    except Exception as e:
+        logging.error(f"Error calculating SHA-256 for {file_path}: {e}")
+        raise
+
+
+def verify_file_sha256(file_path: str, expected_sha256: str, chunk_size: int = 4096) -> bool:
+    """Verify that a file's SHA-256 matches the expected value."""
+    if not os.path.exists(file_path):
+        logging.error(f"File not found for SHA verification: {file_path}")
+        return False
+
+    try:
+        actual_sha256 = calculate_file_sha256(file_path, chunk_size)
+        is_valid = actual_sha256 == expected_sha256
+
+        if is_valid:
+            logging.info(f"SHA-256 verification successful for {os.path.basename(file_path)}")
+        else:
+            logging.error(f"SHA-256 verification failed for {os.path.basename(file_path)}")
+            logging.error(f"Expected: {expected_sha256}")
+            logging.error(f"Actual:   {actual_sha256}")
+
+            # Log to DataDog for monitoring and alerting
+            try:
+                from .logging_utilities import log_as_json
+
+                log_as_json(
+                    {
+                        "event": "sha256_verification_failed",
+                        "file_name": os.path.basename(file_path),
+                        "file_path": file_path,
+                        "expected_sha256": expected_sha256,
+                        "actual_sha256": actual_sha256,
+                        "error_type": "checksum_mismatch",
+                    }
+                )
+            except ImportError:
+                logging.warning("Could not import log_as_json for DataDog logging")
+
+        return is_valid
+    except Exception as e:
+        logging.error(f"Error during SHA-256 verification for {file_path}: {e}")
+
+        # Log to DataDog for monitoring and alerting
+        try:
+            from .logging_utilities import log_as_json
+
+            log_as_json(
+                {
+                    "event": "sha256_verification_error",
+                    "file_name": os.path.basename(file_path),
+                    "file_path": file_path,
+                    "error_message": str(e),
+                    "error_type": "verification_exception",
+                }
+            )
+        except ImportError:
+            logging.warning("Could not import log_as_json for DataDog logging")
```

Comment:
```suggestion
            logging.warning("Could not import log_as_json")
```
question: is  `log_as_json()` uniquely sending messages to Datadog? i had been under the impression that all app logging generally goes to Datadog, and the benefit of `log_as_json()`  was for convenience in formatting the json payload. i could be wrong. Do you mind double-checking this?